omod/src/main/java/org/openmrs/web/rest/resource/TagResource.java

```diff
@@ -0,0 +1,138 @@
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+/**
+ */
+@Resource(name = RestConstants.VERSION_1 + "/tag", supportedClass = Tag.class, supportedOpenmrsVersions = { "1.11.*",
+        "1.12.*", "2.0.*" })
+public class TagResource extends DelegatingCrudResource<Tag> {
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceHandler#newDelegate()
+	 */
+	@Override
+	public Tag newDelegate() {
+		return new Tag();
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#getByUniqueId(String)
+	 */
+	@Override
+	public Tag getByUniqueId(String uuid) {
+		return getService().getTagByUuid(uuid);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceHandler#save(Object)
+	 */
+	@Override
+	public Tag save(Tag tag) {
+		return getService().saveTag(tag);
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#delete(Object,String, RequestContext)
+	 */
+	@Override
+	protected void delete(Tag tag, String reason, RequestContext context) {
+		TagService tagService = Context.getService(TagService.class);
+		tagService.purgeTag(tag);
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#purge(Object, RequestContext)
+	 */
+	@Override
+	public void purge(Tag tag, RequestContext context) {
+		if (tag != null) {
+			getService().purgeTag(tag);
+		}
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#getRepresentationDescription(Representation)
+	 */
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = null;
+		if (rep instanceof DefaultRepresentation || rep instanceof RefRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("tag");
+			description.addProperty("uuid");
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+		} else if (rep instanceof FullRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("tag");
+			description.addProperty("objectType");
+			description.addProperty("objectUuid");
+			description.addProperty("uuid");
```

Comment:
You need to include the audit info (creator and date created) in the full representation